### ISSUE TITLE
BUG: list-links macro - Link breaking if caption includes CamelCase

### ISSUE THREAD

**Issue by 00SS** [Jermolene/TiddlyWiki5#3771]
On tiddlywiki.com edit the tiddler [Hidden Setting: Search AutoFocus](https://tiddlywiki.com/#Hidden%20Setting%3A%20Search%20AutoFocus)

Add a caption field with value: **`Search AutoFocus`**

Open tiddler: [Hidden Settings](https://tiddlywiki.com/#Hidden%20Settings) that uses `<<list-links "[tag[Hidden Settings]]">>`

Note that the link to the tiddler looks like one link, but it breaks into two values.
**`Search `**  and the space following it links properly.
**`AutoFocus`** becomes a separate CamelCase link.

**Comment by 00SS** [comment]
Just as an idea, this seems to address 2 issues in the `<<list-links ... >>` macro
- Fixes this issue's CamelCase broken link
- Fixes #3753 if used in the draggable also

Of course it is possible that it may cause problems in other ways.

```
\define unCamelCase(undo)
\rules except wikilink
$undo$
\end

\define list-links(filter,type:"ul",subtype:"li",class:"",emptyMessage)
<$type$ class="$class$">
<$list filter="$filter$" emptyMessage=<<__emptyMessage__>>>
<$subtype$>
<$link to={{!!title}}>
<$list filter="[all[current]has[caption]]" 
      emptyMessage="""<$view field="title"/>""">
      <$macrocall $name="unCamelCase" undo={{!!caption}}/>
</$list>
</$link>
</$subtype$>
</$list>
</$type$>
\end
```

**Comment by Jermolene** [comment]
Hi @00SS by convention, the `caption` field is wikified. Thus if you include a camelcase link in a caption field it will be rendered as a link. It's done this way so that we can have caption fields that transclude language tiddlers for translations.

**Comment by 00SS** [comment]
What I can see happening here is that:
 - The macro doesn't just "display" the `caption` field
  - A `caption` field containing some CamelCase is transcluded, making a string that includes CamelCase link(s)
 - The string containing CamelCase link(s) get used inside the `<$link>` widget, creating what looks like one normal link
 - The non-CamelCase parts of the link, both before and after any CamelCase portion, link normally to the tiddler's  `title` field
 - The CamelCase parts link to each **CamelCase's tiddler** `title`
 - The whole link looks like one normal link, but it isn't. You open different tiddlers depending on where in the link you click.

Perhaps a note can be added to the `<<list-links>>` macro explaining that the link created will have this behaviour if a caption contains CamelCase.

**Comment by Jermolene** [comment]
Perhaps it's not so much about the `<<list-links>>` macro as the `caption` field?

**Comment by 00SS** [comment]
I notice that the toc macros in `$:/core/macros/toc` do not show this CamelCase behaviour even though it uses similar code in the `toc-caption` macro. 

The difference is that the transclusion portion is wrapped with: `<$set name="tv-wikilinks" value="no">  . . . </$set>`

What does this do? Can it be done here too?

**Comment by Jermolene** [comment]
Hi @00SS I'd forgotten about that, yes I think the same technique could be used here.

**Comment by 00SS** [comment]
I'd appreciate if you let me try and fix it and make a pull request myself that goes into the core!

**Comment by Jermolene** [comment]
Great thank you @00SS I'd be delighted!

**Comment by 00SS** [comment]
Should be fixed in Pull Request #3772
